mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollection.java

```diff
@@ -113,9 +114,9 @@ private void handleSpecificHeader(String headerName, String headerValue) {
             }
         }
 
-        private void manageAddressField(String headerName, String headerValue) {
+        private void manageAddressField(String headerName, String rawHeaderValue) {
             LenientAddressParser.DEFAULT
-                .parseAddressList(headerValue)
+                .parseAddressList(rawHeaderValue)
```

Comment:
I'm almost sure it's not a good solution: the value can be folded, it can use any header encoding (including Q encoding).
I doubt parseAddressList handle those cases

Reply:
> I doubt parseAddressList handle those cases

Yes I think it can. By debugging on the master branch, I see the value passed in `LenientAddressParser.DEFAULT.parseAddressList()` is the decoded value, which means `LenientAddressParser.DEFAULT.parseAddressList("Üsteliğhan Maşrapa")`. Then this statement returns a string in wrong format: `Üstelihan Ma_rapa`. 

Therefor I made `LenientAddressParser.DEFAULT.parseAddressList()` parse the encoded value which is `=?UTF-8?Q?=C3=9Csteli=C4=9Fhan_Ma=C5=9Frapa?=`

Reply:
And what about folded values ?

Reply:
there is already a test case named `foldedHeaderShouldBeSupported`. And this fix passes that test